lib/pbench/server/api/resources/query_apis/query_results.py

```diff
@@ -0,0 +1,289 @@
+        """
+        POST for Pbench datasets from a specified controller within the
+        specified date range, and owned by a specified user:
+
+        {
+            "user": "email-tag",
+            "controller": "controller-name",
+            "start": "start-time",
+            "end": "end-time"
+        }
+
+        "user" specifies the owner of the data to be searched; it need not
+        necessarily be the user represented by the session token header,
+        assuming the session user is authorized to view "user"s data. This
+        will be used to constrain the Elasticsearch query.
+
+        TODO: When we have authorization infrastructure, we'll need to
+        check that "session user" has rights to view "user" data. We might
+        also default a missing "user" JSON field with the authorization
+        token's user. (Or with the default user "public" if there's no
+        token, indicating there's no logged-in user.)
+
+        "controller" is the name of a Pbench agent controller (normally a host
+        name).
+
+        "start" is a time string representing the starting range of dates
+        to search.
+
+        "end" is a time string representing the end range of dates to
+        to search.
+
+        The Elasticsearch URL string is also dependent upon the configured
+        "prefix". This is a Pbench artifact that allows multiple Pbench
+        instances to share a single Elasticsearch server, by segregating
+        their data into unique namespaces. Each Pbench index name is
+        qualified by the index_prefix of the Pbench server that originally
+        indexed the data. The prefix comes from the pbench-server.cfg
+        file for the server instance. For example, production, "staging",
+        and development server instances might all share one Elasticsearch
+        cluster by choosing unique prefixes to generate unique index
+        names.
+
+        Required headers include
+
+        X-auth-token:   Pbench session authorization token authorized to access
+                        "user"'s data. E.g., "user", "admin" or a user with
+                        whom the dataset has been shared.
+        Content-Type:   application/json
+        Accept:         application/json
+
+        Return payload is a summary of the "aggregations" property of the
+        returned Elasticsearch query results, showing fields from the "run"
+        subdocument, including the controller and name, the start and end
+        time, and a starting UNIX timestamp.
+
+        TODO: We probably need to return all *unowned* runs when called
+        without a session token or user parameter. We need to define precisely
+        how this should work.
+
+        NOTE: This is the format currently constructed by the Pbench
+        dashboard `src/model/dashboard.js` fetchResults method, which
+        becomes part of the Redux state. (Note that the "key" of the sequence
+        is the same as the "run.name".)
```

Comment:
I'm not sure what you mean by "size of the ES indices" since this is about post-processing the ES query results into what the dashboard code expects.

If you're talking about the duplicated `"key"` value, that's a Javascript thing; a React component that includes an array of data will get lint-dinged if there's no `key`. But it has nothing to do with Elasticsearch...

Reply:
Sorry -- I forgot where we were.  If we're doubling up values in the object being sent to JS, that's OK (presumably there won't be so much that it impacts network performance).

My concern was that we were doubling up values in the ES documents:  it doesn't hurt much there until you have millions and millions of documents, and then it starts to seriously impact the on-disk representation of the ES index.